### ISSUE TITLE
xNVSE inline variable declarations comes up as error in error checking.

### ISSUE THREAD

**Issue by AVeryUncreativeUsername** [TES5Edit/TES5Edit#974]
**What is the version of xEdit you are using?**
FNVEdit 4.1.3j EXTREMELY EXPERIMENTAL (RC2 for 4.1.4) (A1B0582B) starting session 2021-11-08 14:53:19

**Describe the bug**
xEdit sees xNVSE's [inline variable declaration](https://geckwiki.com/index.php?title=Inline_variable_declaration) as an error when using the "Check For Errors" function. 

This is an example of a script from one of my mods using inline variable declarations. 
https://pastebin.com/HPUHERJf

This causes a false error to come up in the view window. I have had a couple false bug reports on my mods as a result. 

**To reproduce**
1. Load up the plugin with inline variable declarations. (My [mod ](https://www.nexusmods.com/newvegas/mods/73242)for example)
2. Right click on the plugin name or script with inline variable declarations and click "Check For Errors"
3. Check in the message window where it shows the script with the inline variables as an error.
 
**Expected behavior**

I feel the expected behavior should be for it to come back with no errors when using xNVSE inline variable declarations. Though I understand that it's a newer xNVSE feature and xEdit hasn't been updated to account for it yet. 

**Screenshots**

Checking for errors, xEdit flags it as an error.
https://i.imgur.com/9BnqsDW.png

When viewing the local variables, it will come up as unknown.
https://i.imgur.com/1GkDQOa.png

**Additional context**
No other context.

**Comment by ElminsterAU** [comment]
@korri123, @jazzisparis, please provide me with technical information about any file contents affecting changes FNVEdit should support.

Please be aware that I plan to push the next release out to NexusMods before Skyrim AE is released, so there is a very limited time window available.

**Comment by korri123** [comment]
This change was unintentional, I forgot that Bethesda marked the type as either int or floatOrRef while I assigned the "real" variable type. The flag is unused in-game and there are mods already compiled with the new type IDs so I guess it's up to you if you want to support it or not in xEdit.
```C
		eVarType_Float = 0,			//ref is also zero
		eVarType_Integer = 1

		// NVSE, return values only
		eVarType_String = 2
		eVarType_Array = 3
		eVarType_Ref = 4

		eVarType_Invalid = 5
```

**Comment by ElminsterAU** [comment]
ok, I'll change the definition from:
```
wbInteger('Flags', itU8, wbFlags(['IsLongOrShort']), cpCritical),
```
to:
```
wbInteger('Type', itU8, wbEnum(['Float', 'Integer', 'String', 'Array', 'Ref', 'Invalid']), cpCritical),
```
That should do it, right? This is for FNV only, right?

**Comment by korri123** [comment]
FNV only, yes. In vanilla the field is actually meant to be a boolean `IsInteger` but now with these changes from NVSE it can mean more unfortunately.

To be more accurate to current, I would define it as
```delphi
wbInteger('Type', itU8, wbEnum(['FloatOrRef', 'Integer', 'String', 'Array', 'Ref', 'Invalid']), cpCritical),
```

Since for vanilla scripts value 0 can mean either float or refs.

**Comment by ElminsterAU** [comment]
yeah, I already made that change in code after posting the above
it's now:
```
    wbInteger('Type', itU8, wbEnum(['FloatOrRef', 'Integer', 'String (xNVSE)', 'Array (xNVSE)', 'Ref (xNVSE)', 'Invalid (xNVSE)']), cpCritical),
```

**Comment by ElminsterAU** [comment]
I also did ping you on the xEdit discord server with a link to a new FNVEdit build with this change to try out.

**Comment by ElminsterAU** [comment]
should be resolved with commits referenced above (included in 4.0.4)
